src/vm_pages.rs

```diff
@@ -1751,6 +1752,100 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         Ok(())
     }
 
+    /// Invalidates a page range.
+    pub fn block_pages(&self, page_addr: GuestPageAddr, len: u64) -> Result<()> {
+        let invalidated = self
+            .inner
+            .root
+            .invalidate_range_sparse(page_addr, len, |addr| {
```

Comment:
Got it! That makes sense!

Reply:
I patched this and then realized the `tellus`/`guestvm` test was broken. Basically what happens is that when the `guestvm` first tries to share some untouched memory, then `tellus` invokes `block_pages()`, but then `salus` only finds `Unused` PTEs from the PT walk.
Do I need to reuse the logic that was present in `share_mem_region_begin()` so that trying to share a range if the page is not mapped would not cause a breaking ecall but instead a resumable one? Or instead, do you think the page should have been mapped by `tellus` beforehand?

Reply:
ah, in the case that the region is totally unpopulated to begin with, the host does not have to do the block/remove sequence since there's nothing to remove. so in that case the host can just put the return value in A0 and run the vCPU again.

Reply:
Alright!